Makefile

```diff
@@ -48,7 +48,8 @@ EXTRACT_XISO = $(NXDK_DIR)/tools/extract-xiso/extract-xiso
 TOOLS        = cxbe vp20compiler fp20compiler extract-xiso
 NXDK_CFLAGS  = -target i386-pc-win32 -march=pentium3 \
                -ffreestanding -nostdlib -fno-builtin -fno-exceptions \
-               -I$(NXDK_DIR)/lib -I$(NXDK_DIR)/lib/xboxrt \
+               -I$(NXDK_DIR)/lib -I$(NXDK_DIR)/lib/xboxrt/libc_extensions \
+               -I$(NXDK_DIR)/lib/hal \
```

Comment:
This include path shouldn't be added in my opinion.

We should force people to explicitly use `hal/<...>.h` so they know which APIs belong to this legacy API.
This is also helpful when we do libraryization as people can explicitly see which libs they use directly.

Unfortunately, at this point, it's still necessary because:
```c
/lib/pdclib/platform/xbox/functions/assert/assert.c:2:10: fatal error: 
      'debug.h' file not found
#include <debug.h>
```

We should fix this in the future.